tests/integration/specification_model_test.py

```diff
@@ -479,3 +480,226 @@ def test_consistency_specification_parsing_generation(tmp_path: Path) -> None:
+                                Msg2: 2048
+          """,
+            "",
+        ),
+        (
+            """Session:
+                Session:
+                    Buffer_Size:
+                        Default: 1024
+                        Local:
+                           Start: {}
+          """,
+            "",
+        ),
+        (
+            """Session:
+                Session:
+                    Buffer_Size:
+                        Default: 1024
+          """,
+            "",
+        ),
+    ],
+)
+def test_rfi_files(tmp_path: Path, rfi_content: str, match_error: str) -> None:
+    p = parser.Parser()
+    content = """package Test is
+   type Message_Type is (MT_Null => 0, MT_Data => 1, MT_Value => 2, MT_Values => 3, MT_Option_Types => 4,
+                         MT_Options => 5, MT_Unconstrained_Data => 6, MT_Unconstrained_Options => 7) with Size => 8;
+
+   type Length is range 0 .. 2 ** 16 - 1 with Size => 16;
+
+   type ValueT is mod 256;
+
+   type Values is sequence of ValueT;
+
+   type Option_Type is (OT_Null => 0, OT_Data => 1) with Size => 8, Always_Valid => True;
+
+   type Option_Types is sequence of Option_Type;
+
+   type Option is
+      message
+         Option_Type : Option_Type
+            then null
+            if Option_Type = OT_Null
+            then Length
+            if Option_Type = OT_Data;
+         Length : Length
+            then Data
+            with Size => Length * 8;
+        Data : Opaque;
+      end message;
+   type Options is sequence of Option;
+   type Message is
+      message
+         Message_Type : Message_Type
+            then null
+            if Message_Type = MT_Null
+            then Data
+            with Size => Message'Last - Message_Type'Last
+            if Message_Type = MT_Unconstrained_Data
+            then Length
+            if Message_Type /= MT_Null
+               and Message_Type /= MT_Unconstrained_Data
+               and Message_Type /= MT_Unconstrained_Options
+            then Options
+            with Size => Message'Last - Message_Type'Last
+            if Message_Type = MT_Unconstrained_Options;
+         Length : Length
+            then Data
+            with Size => Length * 8
+            if Message_Type = MT_Data
+            then Option_Types
+            with Size => Length * 8
+            if Message_Type = MT_Option_Types
+            then Options
+            with Size => Length * 8
+            if Message_Type = MT_Options
+            then Value
+            if Message_Type = MT_Value
+               and Length = 1
+            then Values
+            with Size => Length * 8
+            if Message_Type = MT_Values;
+         Data : Opaque
+            then null;
+         Option_Types : Option_Types
+            then null;
+         Options : Options
+            then null;
+         Value : ValueT
+            then null;
+         Values : Values;
+      end message;
+
+   for Message use (Data => Option);
```

Comment:
Here it would make sense to use a simpler message type instead of using a copy of `Universal::Message`.

Reply:
I used a simpler message in d095c573a1defa433803ca5da716b75140cbca52